File: generator/src/test/resources/swift/expected/RequestUriParamsTest/test-optional-uri-parameter-generation.output.swift
import Sunday

public final class API<TransportType : Transport> : Sendable {

  public static var problemTypes: [ProblemRegistration] {
    return []
  }
  public let transport: TransportType
  public let defaultContentTypes: [MediaType]
  public let defaultAcceptTypes: [MediaType]

  public init(
    transport: TransportType,
    defaultContentTypes: [MediaType] = [],
    defaultAcceptTypes: [MediaType] = [.json],
    problemTypes: [ProblemRegistration] = API.problemTypes
  ) {
    self.transport = transport
    self.defaultContentTypes = defaultContentTypes
    self.defaultAcceptTypes = defaultAcceptTypes
    problemTypes.forEach { $0.register(on: transport) }
  }

  public func fetchTest(
    def2: Int? = 10,
    obj: Test? = nil,
    str: String? = nil,
    def1: String? = "test",
    int: Int? = nil,
    def: String
  ) throws -> Sunday.Operation<Empty, Test, TransportType> {
    return Sunday.Operation(
      transport: self.transport,
      spec: Sunday.OperationSpec(
        method: .get,
        pathTemplate: "/tests/{obj}/{str}/{int}/{def}/{def1}/{def2}",
        pathParameters: [
          "def2": try ParameterValues.encode(def2),
          "obj": try ParameterValues.encode(obj),
          "str": try ParameterValues.encode(str),
          "def1": try ParameterValues.encode(def1),
          "int": try ParameterValues.encode(int),
          "def": try ParameterValues.encode(def)
        ].filter { $0.value != nil },
        queryParameters: nil,
        body: Empty.none,
        contentTypes: nil,
        acceptTypes: self.defaultAcceptTypes,
        headers: nil
      )
    )
  }

}

File: generator/src/test/resources/kotlin/expected/sunday/RequestUriParamsTest/test-optional-uri-parameter-generation.output.kt
package io.test.service

import io.outfoxx.sunday.MediaType
import io.outfoxx.sunday.RequestFactory
import io.outfoxx.sunday.http.Method
import io.test.Test
import kotlin.Int
import kotlin.String
import kotlin.collections.List

public class API(
  public val requestFactory: RequestFactory,
  public val defaultContentTypes: List<MediaType> = listOf(),
  public val defaultAcceptTypes: List<MediaType> = listOf(MediaType.JSON),
) {
  public suspend fun fetchTest(
    def2: Int? = 10,
    obj: Test? = null,
    str: String? = null,
    def1: String? = "test",
    int: Int? = null,
    def: String,
  ): Test = this.requestFactory
    .result(
      method = Method.Get,
      pathTemplate = "/tests/{obj}/{str}/{int}/{def}/{def1}/{def2}",
      pathParameters = mapOf(
        "def2" to def2,
        "obj" to obj,
        "str" to str,
        "def1" to def1,
        "int" to int,
        "def" to def
      ).filterValues { it != null },
      acceptTypes = this.defaultAcceptTypes
    )
}
